packages/fetchai/connections/ledger/contract_dispatcher.py

```diff
@@ -130,6 +132,30 @@ def get_error_message(
         dialogue.update(response)
         return response
 
+    def _handle_request(
```

Comment:
this is the method that does all the work; it is parametrized to save some line of code.